src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -72,7 +72,7 @@ export const useSearch = (services: DiscoverViewServices) => {
   const [savedSearch, setSavedSearch] = useState<SavedSearch | undefined>(undefined);
   const { savedSearch: savedSearchId, sort, interval } = useSelector((state) => state.discover);
   const { data, filterManager, getSavedSearchById, core, toastNotifications, chrome } = services;
-  const indexPattern = useIndexPattern(services);
+  const { indexPattern, dataSource } = useIndexPattern(services);
```

Comment:
@kavilla I was using the index pattern hook to pass along the data source information as a temporary solution, should I create another hook for datasource or generalize this index pattern hook?